marlowe-playground-client/src/Text/Parsing/Parser/Basic.purs

```diff
@@ -50,14 +65,21 @@ lines = map String.fromCharArray <<< go <<< String.toCharArray
   break :: forall a. (a -> Boolean) -> Array a -> { init :: (Array a), rest :: (Array a) }
   break p = Array.span (not <<< p)
 
-replaceInPosition :: Position -> Position -> String -> String -> String
-replaceInPosition (Position start) (Position end) value initial =
-  unlines
-    $ mapWithIndex
-        ( \i line ->
-            if i == (start.line - 1) then
-              (take (start.column - 1) line) <> value <> (drop (end.column - 1) line)
-            else
-              line
-        )
-    $ lines initial
+replaceInPosition :: Pos -> Pos -> String -> String -> String
```

Comment:
I think you should consider making this take a record. Too many positional arguments of the same type for my liking.

Reply:
done